src/position.hh

```diff
@@ -76,6 +76,7 @@ namespace std {
 template <> struct hash<ccls::Range> {
   std::size_t operator()(ccls::Range x) const {
     union U {
+      U() {} // On VS2017, still need constructor here
```

Comment:
```
    union U {
      U() {}
      ccls::Range range;
      uint64_t u64;
    } u;
```

Reply:
No comment is needed

Reply:
You may remove constryctor if you will swapunion fields.

Reply:
> You may remove constryctor if you will swapunion fields.

What's the rule applied :)

Reply:
In general thats because defaul definition is not trivially constructible. You may try to add constexpr or remove initialization from Pos or swap union members.
In that case union will construct uint instead of Range. Please see https://ideone.com/iiHf5q